cmd/thanos/rule.go

```diff
@@ -1081,3 +1090,16 @@ func validateTemplate(tmplStr string) error {
 	}
 	return nil
 }
+
+// Filter out PromQL related warnings from warning response and keep store related warnings only.
+func filterOutPromQLWarnings(warns []string, logger log.Logger, query string) []string {
+	storeWarnings := make([]string, 0, len(warns))
+	for _, warn := range warns {
+		if strings.Contains(warn, promQLInfo) || strings.Contains(warn, promQLWarning) {
```

Comment:
We could use https://github.com/thanos-io/thanos/blob/e155196618a3246c0303fea8c02761ffc5f61f76/pkg/extannotations/annotations.go#L12 here or move the function to that package maybe? Er need similar logic in querier to decide if we want to cache a response based on warnings.

Reply:
Nice, didn't know that we have this function. I will use that instead

Reply:
Fixed in latest commit